File: backend/schemas/web_admission.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class WebAdmissionCreate(BaseModel):
    student_name: str = Field(..., min_length=2, max_length=255)
    father_name: str = Field(..., min_length=2, max_length=255)
    phone: str = Field(..., min_length=7, max_length=20)
    email: str | None = None
    address: str | None = None
    admission_category: Literal['tuition', 'professional']
    class_applying: str | None = None
    course_name: str | None = None
    preferred_batch: str | None = None
    previous_school: str | None = None
    message: str | None = None


class WebAdmissionResponse(BaseModel):
    id: int
    student_name: str
    father_name: str
    phone: str
    email: str | None = None
    address: str | None = None
    admission_category: str
    class_applying: str | None = None
    course_name: str | None = None
    preferred_batch: str | None = None
    previous_school: str | None = None
    message: str | None = None
    isActive: bool
    created_at: datetime
    updated_at: datetime | None = None

    class Config:
        from_attributes = True

File: backend-express/src/models/WebAdmission.js
import mongoose from 'mongoose';

const webAdmissionSchema = new mongoose.Schema(
  {
    student_name: { type: String, required: true, trim: true, minlength: 2, maxlength: 255 },
    father_name: { type: String, required: true, trim: true, minlength: 2, maxlength: 255 },
    phone: { type: String, required: true, trim: true, minlength: 7, maxlength: 20 },
    email: { type: String },
    address: { type: String },
    admission_category: { type: String, enum: ['tuition', 'professional'], required: true },
    class_applying: { type: String },
    course_name: { type: String },
    preferred_batch: { type: String },
    previous_school: { type: String },
    message: { type: String },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' } }
);

webAdmissionSchema.index({ isActive: 1, created_at: -1 });

export const WebAdmission = mongoose.model('WebAdmission', webAdmissionSchema);
